python/ray/train/mosaic/mosaic_predictor.py

```diff
@@ -0,0 +1,60 @@
+from ray.train.torch import TorchPredictor
+from ray.train.batch_predictor import BatchPredictor
+from ray.air.checkpoint import Checkpoint
+from pathlib import Path
+import torch
+
+
+class MosaicPredictor(BatchPredictor):
+    """A BatchPredictor wrapper for Mosaic's Composer models.
+
+    The state dicts saved by composer `CheckpointSaver`s have extra
+    wrapping around normal PyTorch models' state dict. As such, it is
+    necessary to unwrap those to properly load models from saved checkpoints.
+    Additionally, existing class methods of BatchPredictor does not allow
+    directly using checkpoint path, but rather takes a ray checkpoint object,
+    this wrapper provides creating a BatchPredictor from a checkpoint path.
+    """
+
+    @classmethod
+    def from_save_path(
+        cls,
+        path: Path,
+        model: torch.nn.Module,
+        strict: bool = False,
+        **predictor_kwargs
+    ):
+        """
+        This function creates a BatchPredictor from a saved Composer checkpoint.
+        Because Composer library is built on PyTorch, `TorchPredictor` is used
+        as the `predictor_cls`
+
+        Args:
+            path: Path to the saved checkpoint object
+            model: the model to which the saved checkpoint will be loaded. This
+                model should be a bare torch model, without any composer wrapping.
+            strict: the boolean variable for strict state_dict loading onto the model
+        """
+        model_state_dict = torch.load(path)["state"]["model"]
+
+        # remove module prefixes when loading the model weights
+        while True:
```

Comment:
isn't there a built-in torch util for this? I may be misremembering though

Reply:
There is but I purposefully did not use it because that function has extra things that is not needed for our case. (Intended to reduce the overhead of having to call another function + spending time doing unnecessary tasks)

Reply:
Is there actually a noticeable performance difference? If not, I'd reuse existing code. We don't need to shave every millisecond :)

Reply:
I have not really measured the performance. However, one advantage of using this code instead of the existing function is that we can easily note whether a prefix has been removed or not. The existing function does not return anything, so we would have to iterate through and see if there are any changes from the previous version.

Reply:
That's fair! let's make sure to add a short comment explaining why we aren't using the built-in function then